addons/pos_razorpay/static/src/app/screens/payment_screen/payment_screen.js

```diff
@@ -23,4 +23,32 @@ patch(PaymentScreen.prototype, {
             }
         });
     },
+    addNewPaymentLine(paymentMethod) {
+        if (paymentMethod.use_payment_terminal === "razorpay" && this.isRefundOrder) {
+            const refundedOrder = this.currentOrder.lines[0]?.refunded_orderline_id?.order_id;
+            const razorpayPaymentlines = refundedOrder.payment_ids.filter(
+                (pi) => pi.payment_method_id.use_payment_terminal === "razorpay"
+            );
+            const transactions_ids = this.currentOrder.payment_ids.map((pi) => pi.transaction_id);
+            const res = super.addNewPaymentLine(paymentMethod);
+            const newPaymentLine = this.paymentLines.at(-1);
+            if (
+                res &&
+                (razorpayPaymentlines.length !== 1 ||
+                    Math.abs(newPaymentLine.amount) < razorpayPaymentlines[0].amount ||
+                    newPaymentLine.amount === 0 ||
```

Comment:
In which case does this line triggers a true and the line above triggers a false ?

Reply:
If a payment line for a refund is added, the response (res) will be true. However, if the refunded order contains a single Razorpay payment line, and the amount of the new payment line is greater than the amount of the Razorpay payment line in the refunded order, while the new payment line amount is not equal to 0, the second condition will be false. In this case, there is no need to delete the added payment line.

Basically, we do not allow the addition of a new payment line if the above conditions are true.